plugins/chat/assets/javascripts/discourse/components/chat-composer.js

```diff
@@ -218,32 +213,33 @@ export default Component.extend(TextareaTextManipulation, {
     this._super(...arguments);
 
     if (
-      !this.editingMessage &&
+      !this.composerService.editingMessage &&
       this.chatChannel?.draft &&
       this.chatChannel?.canModifyMessages(this.currentUser)
     ) {
       // uses uploads from draft here...
-      this.setProperties({
-        value: this.chatChannel.draft.message,
-        replyToMsg: this.chatChannel.draft.replyToMsg,
-      });
+      this.set("value", this.chatChannel.draft.message);
+      this.composerService.setReplyTo(this.chatChannel.draft.replyToMsg);
 
       this._captureMentions();
       this._syncUploads(this.chatChannel.draft.uploads);
-      this.setInReplyToMsg(this.chatChannel.draft.replyToMsg);
     }
 
-    if (this.editingMessage && !this.loading) {
-      this.setProperties({
-        replyToMsg: null,
-        value: this.editingMessage.message,
-      });
+    this.resizeTextarea();
+  },
+
+  // FIXME (martin) Is this okay to do?? Can't use didReceiveAttrs because now
+  // livePanel.composer.editingMessage is changing not a scoped attribute for this component.
+  @observes("composerService.editingMessage")
```

Comment:
you can do: `{{did-update this.updateEditingMessage this.composerService.editingMessage}}` on a node of the chat-composer-template